datumaro/plugins/celeba_format.py

```diff
@@ -0,0 +1,151 @@
+            image_dir = osp.join(osp.dirname(self._anno_dir), CelebaPath.IMAGES_ALIGN_DIR)
+        if osp.isdir(image_dir):
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace('\\', '/'): p
+                for p in find_images(image_dir, recursive=True)
+            }
+        elif osp.isdir(osp.join(osp.dirname(self._anno_dir), CelebaPath.IMAGES_DIR)):
+            images = {}
+
+        label_categories = self._categories[AnnotationType.label]
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                item_id, item_ann = self.split_annotation(line)
+                label_ids = [int(id) for id in item_ann]
+                anno = []
+                for label in label_ids:
+                    if label > len(label_categories):
+                        for i in range(len(label_categories), label + 1):
+                            label_categories.add('label_%d' % i)
+                    anno.append(Label(label))
+
+                items[item_id] = DatasetItem(id=item_id, image=images.get(item_id),
+                    annotations=anno)
+
+        bbox_path = osp.join(self._anno_dir, CelebaPath.BBOXES_FILE)
+        if osp.isfile(bbox_path):
+            with open(bbox_path, encoding='utf-8') as f:
+                for line in f:
+                    if len(line.split()) < 5 or 'image_id' == line[0:8]:
+                        continue
+                    item_id, item_ann = self.split_annotation(line)
+                    bbox = [float(id) for id in item_ann]
+                    if item_id not in items:
+                        items[item_id] = DatasetItem(id=item_id, image=images.get(item_id))
+                    anno = items[item_id].annotations
+                    label = None
+                    if 0 < len(anno):
+                        label = anno[0].label
+                    anno.append(Bbox(bbox[0], bbox[1], bbox[2], bbox[3], label=label))
+
+        attr_path = osp.join(self._anno_dir, CelebaPath.ATTRS_FILE)
+        if osp.isfile(attr_path):
+            with open(attr_path, encoding='utf-8') as f:
+                f.readline()
+                line = f.readline()
+                name_attr = line.split()
+                attr = {}
+                for line in f:
+                    item_id, item_ann = self.split_annotation(line)
+                    if len(name_attr) != len(item_ann):
+                        continue
+                    for i, ann in enumerate(item_ann):
+                        attr[name_attr[i]] = ann
+
+                    if item_id not in items:
+                        items[item_id] = DatasetItem(id=item_id, image=images.get(item_id))
+                    items[item_id].attributes = attr
+
+        landmark_path = osp.join(self._anno_dir, CelebaPath.LANDMARKS_FILE)
+        if not osp.isfile(landmark_path):
+            landmark_path = osp.join(self._anno_dir, CelebaPath.LANDMARKS_ALIGN_FILE)
```

Comment:
There probably should be a check that we use `LANDMARKS_ALIGN_FILE` if and only if we are using `IMAGES_ALIGN_DIR`, since mixing the aligned images with non-aligned landmarks (or vice versa) would result in complete nonsense.

And now that I think about it, if we load aligned images, we shouldn't load boxes at all, since that would also result in nonsensical annotations.